src/app/review.service.ts

```diff
@@ -21,9 +21,25 @@ export class ReviewService {
     this.reviewChange.next();
   }
 
-  getByMovieId(movieId: string): Observable<Review[]> {
+  getReviewsPage(movieId: string, pagination: ReviewPagination): Observable<ReviewsPage> {
+
+    const params = {
+      sortBy: pagination.sortBy,
+      direction: pagination.direction,
+      limit: pagination.limit + '',
+      page: pagination.page + '',
+    };
+
+    if (pagination.filter.author) {
+      params['filterAuthorId'] = pagination.filter.author.id;
+    }
+
+    if (pagination.filter.rating) {
+      params['filterRating'] = pagination.filter.rating + '';
+    }
+
     return this.reviewChange.asObservable().pipe(
-      mergeMap(() => this.http.get<Review[]>(`${environment.apiUrl}/movies/${movieId}/reviews`))
+      mergeMap(() => this.http.get<ReviewsPage>(`${environment.apiUrl}/movies/${movieId}/reviews`, { params }))
```

Comment:
Previously we had a review array. how is it that 'ReviewsPage' now takes over as non-Array?

Reply:
The array is still inside `ReviewsPage` as a attribute named `reviews`.